easybuild/tools/filetools.py

```diff
@@ -460,26 +463,49 @@ def download_file(filename, url, path, forced=False):
     attempt_cnt = 0
 
     # use custom HTTP header
-    url_req = urllib2.Request(url, headers={'User-Agent': 'EasyBuild',  "Accept" : "*/*"})
+    headers = {'User-Agent': 'EasyBuild', 'Accept': '*/*'}
+    if not HAVE_REQUESTS:
+        url_req = urllib2.Request(url, headers=headers)
 
     while not downloaded and attempt_cnt < max_attempts:
         try:
-            # urllib2 does the right thing for http proxy setups, urllib does not!
-            url_fd = urllib2.urlopen(url_req, timeout=timeout)
-            _log.debug('response code for given url %s: %s' % (url, url_fd.getcode()))
+            if HAVE_REQUESTS:
+                url_req = requests.get(url, headers=headers, stream=True, timeout=timeout)
+                status_code = url_req.status_code
+                url_req.raise_for_status()
+                url_fd = url_req.raw
+                url_fd.decode_content = True
+            else:
+                # urllib2 does the right thing for http proxy setups, urllib does not!
+                url_fd = urllib2.urlopen(url_req, timeout=timeout)
+                status_code = url_fd.getcode()
+            _log.debug('response code for given url %s: %s' % (url, status_code))
```

Comment:
@bartoldeman Hmm, this whole block is becoming a bit much now, especially since `download_file` is already quite messy... Can we move this up into a dedicated function, something like:

```python
status_code = download_from_url_to(...)
```